arangod/IResearch/IResearchFeature.cpp

```diff
@@ -144,6 +144,20 @@ typedef arangodb::aql::AqlValue (*IResearchFunctionPtr)(
   arangodb::SmallVector<arangodb::aql::AqlValue> const&
 );
 
+size_t computeThreadPoolSize(size_t threads, uint32_t threadsLimit) {
+  static const unsigned int MAX_THREADS = 8; // arbitrary limit on the upper bound of threads in pool
+  static const unsigned int MIN_THREADS = 1; // at least one thread is required
+  auto maxThreads = threadsLimit ? threadsLimit : MAX_THREADS;
+
+  return threads
+    ? threads
+    : std::max(
```

Comment:
Using different types for `std::min` or `std::max` is like asking for trouble.
At least the compilers on MacOS may be picky about it if you mix `size_t`, `uint32_t` or `unsigned int`. These types may boil down to the same fundamental type or not.
It is safest and easiest to understand to use a single data type for the number of threads (either `size_t`, `uint32_t` or `unsigned int`, but not 3 different ones.